api/envoy/extensions/grpc_credentials/file_based_metadata.proto

```diff
@@ -0,0 +1,26 @@
+syntax = "proto3";
+
+// [#protodoc-title: Grpc Credentials File Based Metadata]
+// Configuration for File Based Metadata Grpc Credentials Plugin
+
+package envoy.extensions.grpc_credentials;
+option go_package = "grpc_file_creds";
+
+import "envoy/api/v2/core/base.proto";
+import "validate/validate.proto";
+
+message FileBasedMetadataConfig {
+
+  // Location or inline data of secret to use for authentication of the Google gRPC connection
+  // this secret will be attached to a header of the gRPC connection
+  envoy.api.v2.core.DataSource secret_data = 1;
+
+  // Metadata header key to use for sending the secret data
+  // if no header key is set, "authorization" header will be used
+  string header_key = 2;
+
+  // Prefix to prepend to the secret in the metadata header
+  // if no prefix is set, the default is to use no prefix
+  string header_prefix = 3;
```

Comment:
Is this (prefix) going to be flexible enough for anticipated future needs?

Reply:
I included this mostly to support the `Authorization: Bearer <token>` style headers. Are there other common patterns that need more configurable options to be supported?

Reply:
IDK; @lizan @incfly ?